momepy/preprocessing.py

```diff
@@ -1606,6 +1612,16 @@ def __init__(
 
         # Store geometries as a GeoDataFrame
         self.polygons = gpd.GeoDataFrame(geometry=polygons)
+
+        if self.polygons.empty:
+            warnings.warn(
+                "Input roads could not not be polygonized. "
+                "Identification of face artifacts not possible.",
+                UserWarning,
+                stacklevel=2,
+            )
```

Comment:
```suggestion
            )
            self.threshold = None
            self.face_artifacts = None
```

I am not sure if we should not do this and maybe some other.

Reply:
So should we go ahead and assign off possible attributes as `None`?

* `kde`
* `pdf`
* `peaks`
* `d_peaks`
* `valleys`
* `d_valleys`
* `threshold`
* `face_artifacts`

Reply:
I went ahead and pushed up all `attr = None`